src/Domain/AvailabilitiesDomain.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain;
+
+use App\Entity\User;
+use App\Entity\UserAvailability;
+use Doctrine\Persistence\ObjectManager;
+
+class AvailabilitiesDomain
+{
+    public array $availabilities;
+
+    public function __construct(\DateTimeImmutable $start, \DateTimeImmutable $end, array $userAvailabilities)
+    {
+        $this->availabilities = [];
+
+        $period = new \DatePeriod($start, new \DateInterval('PT2H'), $end);
+
+        foreach ($period as $date) {
+            $userAvailability = null;
+            foreach ($userAvailabilities as $k => $ua) {
+                if ($ua->startTime->format('Y-m-d H:i') === $date->format('Y-m-d H:i')) {
+                    $userAvailability = $ua;
+                    unset($userAvailabilities[$k]);
+
+                    break;
+                }
+            }
+
+            $this->availabilities[] = new AvailabilityDomain($date, $userAvailability);
+        }
+    }
+
+    public function compute(ObjectManager $om, User $user): void
+    {
+        foreach ($this->availabilities as $availability) {
+            if ($availability->tick && null === $availability->userAvailability) {
+                $om->persist(new UserAvailability(
+                    null,
+                    $user,
+                    $availability->date,
+                    $availability->date->add(new \DateInterval('PT2H')),
+                    UserAvailability::STATUS_AVAILABLE
+                ));
+            } elseif (!$availability->tick && null !== $availability->userAvailability) {
+                $om->remove($availability->userAvailability);
+            }
+        }
+
+        $om->flush();
+    }
+}
+
+class AvailabilityDomain
```

Comment:
:thinking: why multiple classes in one file ?

Reply:
Because this class is usable only with AvailibilitiesDomain context.